File: app/src/main/java/com/tammamkhalaf/myuaeguide/databases/hereDeveloper/AppDatabase.java
package com.tammamkhalaf.myuaeguide.databases.hereDeveloper;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.ItemConverter;
import com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.PlaceResponse;

@Database(entities = PlaceResponse.class, version = 1)
@TypeConverters(ItemConverter.class)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase instance;
    public abstract PlacesDao placesDao();

    public static synchronized AppDatabase getInstance(Context context) {

        if(instance == null){
            instance = Room.databaseBuilder(context, AppDatabase.class,"AppDatabase")
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return instance;
    }
}

File: shared/src/commonMain/kotlin/com/tammamkhalaf/kmmsharedmodule/databases/hereDeveloper/AppDatabase.kt
package com.tammamkhalaf.myuaeguide.databases.hereDeveloper

import androidx.room.Database

@Database(entities = PlaceResponse::class, version = 1)
@TypeConverters(ItemConverter::class)
abstract class AppDatabase : RoomDatabase() {
    abstract fun placesDao(): com.tammamkhalaf.myuaeguide.databases.hereDeveloper.PlacesDao?

    companion object {
        private var instance: AppDatabase? = null
        @Synchronized
        fun getInstance(context: android.content.Context?): AppDatabase? {
            if (instance == null) {
                instance = Room.databaseBuilder(context, AppDatabase::class.java, "AppDatabase")
                        .fallbackToDestructiveMigration()
                        .build()
            }
            return instance
        }
    }
}